TrkDiag/src/BkgDiag_module.cc

```diff
@@ -474,22 +474,20 @@ namespace mu2e
     // find the momentum for the first step point from the primary particle in this delta
     for(auto id : dids) {
       StrawDigiMC const& mcdigi = _mcdigis->at(id);
-      // use TDC channel 0 to define the MC match
-      auto const& spmcp = mcdigi.earlyStrawGasStep();
-      art::Ptr<SimParticle> const& spp = spmcp->simParticle();
-
+      auto const& sgsp = mcdigi.earlyStrawGasStep();
+      art::Ptr<SimParticle> const& spp = sgsp->simParticle();
       if(spp == pptr){
-        pmom = sqrt(spmcp->momentum().mag2());
+        pmom = sqrt(sgsp->momentum().mag2());
         break;
       }
     }
   }
 
   void BkgDiag::fillStrawHitInfoMC(StrawDigiMC const& mcdigi, art::Ptr<SimParticle>const& pptr, StrawHitInfo& shinfo) const {
     // use TDC channel 0 to define the MC match
-    auto const& spmcp = mcdigi.earlyStrawGasStep();
-    art::Ptr<SimParticle> const& spp = spmcp->simParticle();
-    shinfo._mct0 = _toff.timeWithOffsetsApplied(*spmcp);
+    auto const& sgsp = mcdigi.earlyStrawGasStep();
```

Comment:
There are a ton of places that could uses an "auto" instead of the variable declaration, this would make the code much more elegant - and I know aristocratic gentlemen like yourself like elegance....